Task2/src/Task2.2/Polynomial.cs

```diff
@@ -0,0 +1,175 @@
+			Polynomial PolynomResult = new Polynomial(сoefPolynomResult);
+			for (int i = 0; i < firstPolynom.СoefPolynom.Length; ++i)
+			{
+				for (int j = 0; j < secondPolynom.СoefPolynom.Length; ++j)
+				{
+					сoefPolynomResult[i + j] += secondPolynom.СoefPolynom[i] * firstPolynom.СoefPolynom[j];
+				}
+			}
+			return PolynomResult;
+		}
+
+		/// <summary>
+		/// The method overrides the mathimatical "plus" operation for working with two polynomials
+		/// </summary>
+		/// <param name="firstPolynom"> the first polynomial</param>
+		/// <param name="secondPolynom">the second polynomial</param>
+		/// <returns>return polynomial's result</returns>
+		public static Polynomial operator +(Polynomial firstPolynom, Polynomial secondPolynom)
+		{
+			if (firstPolynom == null)
+			{
+				throw new NullReferenceException();
+			}
+			if (secondPolynom == null)
+			{
+				throw new NullReferenceException();
+			}
+
+			int degreePolynomResult = 0;
+			if (firstPolynom.СoefPolynom.Length > secondPolynom.СoefPolynom.Length)
+			{
+				degreePolynomResult = firstPolynom.СoefPolynom.Length;
+			}
+			else if (firstPolynom.СoefPolynom.Length <= secondPolynom.СoefPolynom.Length)
+			{
+				degreePolynomResult = firstPolynom.СoefPolynom.Length;
+			}
+			double[] koefPolynomResult = new double[degreePolynomResult];
+
+			Polynomial PolynomResult = new Polynomial(koefPolynomResult);
+
+			for (int i = 0; i < degreePolynomResult; i++)
+			{
+				PolynomResult.СoefPolynom[i] = firstPolynom.СoefPolynom[i] + secondPolynom.СoefPolynom[i];
+			}
+
+			return PolynomResult;
+		}
+
+		/// <summary>
+		/// The method overrides the mathimatical "minus" operation for working with two polynomials
+		/// </summary>
+		/// <param name="firstPolynom"> the first polynomial</param>
+		/// <param name="secondPolynom">the second polynomial</param>
+		/// <returns>return polynomial's result</returns>
+		public static Polynomial operator -(Polynomial firstPolynom, Polynomial secondPolynom)
+		{
+			if (firstPolynom == null)
+			{
+				throw new NullReferenceException();
+			}
+			if (secondPolynom == null)
+			{
+				throw new NullReferenceException();
+			}
+
+			int degreePolynomResult = 0;
+			if (firstPolynom.СoefPolynom.Length > secondPolynom.СoefPolynom.Length)
+			{
+				degreePolynomResult = firstPolynom.СoefPolynom.Length;
+			}
+			else if (firstPolynom.СoefPolynom.Length <= secondPolynom.СoefPolynom.Length)
+			{
+				degreePolynomResult = firstPolynom.СoefPolynom.Length;
+			}
+
+			double[] koefPolynomResult = new double[degreePolynomResult];
+
+			Polynomial PolynomResult = new Polynomial(koefPolynomResult);
+
+			for (int i = 0; i < degreePolynomResult; i++)
+			{
+				PolynomResult.СoefPolynom[i] = firstPolynom.СoefPolynom[i] - secondPolynom.СoefPolynom[i];
+			}
+
+			return PolynomResult;
+		}
+
+		public override bool Equals(object obj)
+		{
+			if (obj!=null)
+			{
+				if (this.GetType()==obj.GetType())
+				{
+					if (obj is Polynomial polynom)
```

Comment:
```suggestion
					Polynomial polynom = (Polynomial)obj;
```
Можно сразу привести тип без условия. т.к. за счет проверки выше мы уверены что тип будет совпадать